docs/cars.py

```diff
@@ -0,0 +1,123 @@
+  all_footnotes = {}
+  i = 1
+  for _, footnotes in get_interface_attr("Footnote").items():
+    if footnotes is not None:
+      for footnote in footnotes:
+        all_footnotes[footnote] = i
+        i += 1
+  return all_footnotes
+
+
+def get_makes_good_torque():
+  makes_good_torque = []
+  for make, good_torque in get_interface_attr("GOOD_TORQUE").items():
+    if good_torque:
+      makes_good_torque.append(make)
+  return makes_good_torque
+
+
+ALL_FOOTNOTES = get_all_footnotes()
+MAKES_GOOD_STEERING_TORQUE = get_makes_good_torque()
+
+
+class CarRow:
+  def __init__(self, car_info, CP):
+    self.make, self.model = car_info.name.split(' ', 1)
+    self.row, star_count = self.get_row(car_info, CP)
+    self.tier = {5: Tier.GOLD, 4: Tier.SILVER}.get(star_count, Tier.BRONZE)
+
+  def get_row(self, car_info, CP):
+    # TODO: add YouTube videos
+    # Some minimum steering speeds are not yet in CarParams
+    min_steer_speed = CP.minSteerSpeed
+    if car_info.min_steer_speed is not None:
+      min_steer_speed = car_info.min_steer_speed
+      assert CP.minSteerSpeed == 0, "Minimum steer speed set in both CarInfo and CarParams for {}".format(CP.carFingerprint)
+
+    min_enable_speed = CP.minEnableSpeed
+    if car_info.min_enable_speed is not None:
+      min_enable_speed = car_info.min_enable_speed
+
+    stars = [CP.openpilotLongitudinalControl and not CP.radarOffCan, min_enable_speed <= 0., min_steer_speed <= 0.,
+             CP.carName in MAKES_GOOD_STEERING_TORQUE, CP.carFingerprint not in non_tested_cars]
+    row = [self.make, self.model, car_info.package, *map(lambda star: "full" if star else "empty", stars)]
+
+    # Check for car footnotes and star demotions
+    star_count = 0
+    for row_idx, column in enumerate(Column):
+      footnote = get_footnote(car_info, column)
+      if column in StarColumns:
+        # Demote if footnote specifies a star
+        if footnote is not None and footnote.value.star is not None:
+          row[row_idx] = footnote.value.star
+        star_count += row[row_idx] == "full"
+        row[row_idx] = get_star_icon(row[row_idx])
+
+      if footnote is not None:
+        row[row_idx] += "[<sup>{}</sup>](#Footnotes)".format(ALL_FOOTNOTES[footnote])
+
+    return row, star_count
+
+
+def get_tiered_cars():
+  # Keep track of cars while sorting by make, model name, and year
+  tiered_cars = {tier: SortedList(key=lambda car: car.make + car.model) for tier in Tier}
+
+  for _, models in get_interface_attr("CAR_INFO").items():
+    for model, car_info in models.items():
+      # Hyundai exception: all have openpilot longitudinal
+      fingerprint = defaultdict(dict)
+      fingerprint[1] = {HKG_RADAR_START_ADDR: 8}
+      CP = interfaces[model][0].get_params(model, fingerprint=fingerprint)
+      # Skip community supported
+      if CP.dashcamOnly:
+        continue
+
+      # Some candidates have multiple variants
+      if not isinstance(car_info, list):
+        car_info = [car_info]
```

Comment:
think this is a bit more clear
```suggestion
        car_info = [car_info, ]
```